docker/upgrade-database.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -euo pipefail
+source ./backup.sh --upgrade
+docker compose down core gui && \
+docker compose down database --volumes && \
+source ./restore.sh --database $backup_dir && \
```

Comment:
Is `source` really necessary? It makes this hard to debug and read (it is unecler where backup_dir comes from for example)

I would suggest something like:

```bash
backup_dir=$(./backup.sh --upgrade)
```

While changing the output of backup.sh to only the backup_dir while running in `UPGRADE_MODE`